src/lib/merkle_tree.ts

```diff
@@ -8,23 +13,45 @@ export { Witness, MerkleTree, MerkleWitness, BaseMerkleWitness };
 type Witness = { isLeft: boolean; sibling: Field }[];
 
 /**
- * Levels are indexed from leafs (level 0) to root (level N - 1).
+ * A [Merkle Tree](https://en.wikipedia.org/wiki/Merkle_tree) is a binary tree in which every leaf is the cryptography hash of a piece of data,
+ * and every node is the hash of the concatenation of its two child nodes.
+ *
+ * A Merkle Tree allows developers to easily and securely verify the integrity of large amounts of data.
+ *
+ * Take a look at our [documentation](https://docs.minaprotocol.com/en/zkapps) on how to use Merkle Trees in combination with zkApps and zero knowledge programming!
+ *
+ * Levels are indexed from leaves (level 0) to root (level N - 1).
  */
 class MerkleTree {
   private nodes: Record<number, Record<string, Field>> = {};
   private zeroes: Field[];
 
+  /**
+   * Creates a new, empty [Merkle Tree](https://en.wikipedia.org/wiki/Merkle_tree).
+   * @param {number} height The height of Merkle Tree.
+   * @returns {MerkleTree} A new MerkleTree
+   */
   constructor(public readonly height: number) {
     this.zeroes = [Field(0)];
     for (let i = 1; i < height; i++) {
       this.zeroes.push(Poseidon.hash([this.zeroes[i - 1], this.zeroes[i - 1]]));
     }
   }
 
+  /**
+   * Returns a node which lives at a given index and level.
+   * @param {number} level Level of the node.
+   * @param {bigint} index Index of the node.
+   * @returns {Field} The data of the node.
```

Comment:
I personally don't like adding types in documentation since TS can infer these types. The comments could drift and disagree with what TS is saying and can be confusing.

Reply:
Good point! didn't think of it that way - its probably better to just leave them out then